packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkNativeInitialization.kt

```diff
@@ -0,0 +1,337 @@
+import com.datadog.android.log.LogsConfiguration
+import com.datadog.android.privacy.TrackingConsent
+import com.datadog.android.rum.RumConfiguration
+import com.datadog.android.rum._RumInternalProxy
+import com.datadog.android.rum.configuration.VitalsUpdateFrequency
+import com.datadog.android.rum.model.ActionEvent
+import com.datadog.android.rum.model.ResourceEvent
+import com.datadog.android.rum.tracking.ActivityViewTrackingStrategy
+import com.datadog.android.telemetry.model.TelemetryConfigurationEvent
+import com.datadog.android.trace.TraceConfiguration
+import com.google.gson.Gson
+import java.util.Locale
+
+/**
+ * Initializes the Android Datadog SDK.
+ */
+class DdSdkNativeInitialization(
+    private val appContext: Context,
+    private val datadog: DatadogWrapper = DatadogSDKWrapper(),
+    private val jsonFileReader: JSONFileReader = JSONFileReader()
+) {
+    internal fun initialize(ddSdkConfiguration: DdSdkConfiguration) {
+        val sdkConfiguration = buildSdkConfiguration(ddSdkConfiguration)
+        val rumConfiguration = buildRumConfiguration(ddSdkConfiguration)
+        val logsConfiguration = buildLogsConfiguration(ddSdkConfiguration)
+        val traceConfiguration = buildTraceConfiguration(ddSdkConfiguration)
+        val trackingConsent = buildTrackingConsent(ddSdkConfiguration.trackingConsent)
+
+        configureSdkVerbosity(ddSdkConfiguration)
+
+        datadog.initialize(appContext, sdkConfiguration, trackingConsent)
+
+        datadog.enableRum(rumConfiguration)
+
+        datadog.enableTrace(traceConfiguration)
+
+        datadog.enableLogs(logsConfiguration)
+    }
+
+    private fun configureSdkVerbosity(configuration: DdSdkConfiguration) {
+        val verbosity =
+            when (configuration.verbosity?.lowercase(Locale.US)) {
+                "debug" -> Log.DEBUG
+                "info" -> Log.INFO
+                "warn" -> Log.WARN
+                "error" -> Log.ERROR
+                else -> null
+            }
+        if (verbosity != null) {
+            datadog.setVerbosity(verbosity)
+        }
+    }
+
+    private fun getDefaultAppVersion(): String {
+        val packageName = appContext.packageName
+        val packageInfo =
+            try {
+                appContext.packageManager.getPackageInfo(packageName, 0)
+            } catch (e: PackageManager.NameNotFoundException) {
+                datadog.telemetryError(e.message ?: DdSdkImplementation.PACKAGE_INFO_NOT_FOUND_ERROR_MESSAGE, e)
+                return DdSdkImplementation.DEFAULT_APP_VERSION
+            }
+
+        return packageInfo?.let {
+            // we need to use the deprecated method because getLongVersionCode method is only
+            // available from API 28 and above
+            @Suppress("DEPRECATION")
+            it.versionName ?: it.versionCode.toString()
+        }
+            ?: DdSdkImplementation.DEFAULT_APP_VERSION
+    }
+
+    @Suppress("ComplexMethod", "LongMethod", "UnsafeCallOnNullableType")
```

Comment:
Any way the `UnsafeCallOnNullableType` can be placed closer to the actual call, or are there multiple?